var/spack/repos/builtin/packages/mpich/package.py

```diff
@@ -162,7 +178,9 @@ def setup_dependent_package(self, module, dependent_spec):
     def autoreconf(self, spec, prefix):
         """Not needed usually, configure should be already there"""
         # If configure exists nothing needs to be done
-        if os.path.exists(self.configure_abs_path):
+        
+        if (os.path.exists(self.configure_abs_path) and
+            not spec.satisfies('@3.3 +hwloc'):
```

Comment:
```suggestion
        if (os.path.exists(self.configure_abs_path) and
            not spec.satisfies('@3.3 +hwloc')):
```
flake8 complains about the empty line and the missing `)`.